modules/apps/portal-configuration/portal-configuration-upgrade-impl/src/main/java/com/liferay/portal/configuration/upgrade/internal/PrefsPropsToConfigurationUpgradeHelperImpl.java

```diff
@@ -133,6 +134,91 @@ else if (returnType == String[].class) {
 		portletPreferences.store();
 	}
 
+	@Override
+	public void mapConfigurationsWithCompanyId(
+			long companyId, Class<?> configurationClass,
+			KeyValuePair... keyValuePairs)
+		throws Exception {
+
+		Dictionary<String, Object> properties = new HashMapDictionary<>();
```

Comment:
@jeffwu0724 The logic here has duplication with existing method, this should be avoided